Tests/tests/test_ScreenCapture.cpp

```diff
@@ -0,0 +1,169 @@
+    FrameBufferApiImplMock frameBufferApiImplMock;
+
+    string response;
+    ServiceMock service;
+    Core::JSONRPC::Message message;
+    FactoriesImplementation factoriesImplementation;
+    PluginHost::IDispatcher* dispatcher;
+
+    ScreenCaptureFixture()
+        : connection(1, 0)
+    {
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+
+    }
+    virtual ~ScreenCaptureFixture()
+    {
+        PluginHost::IFactories::Assign(nullptr);
+    }
+
+    virtual void SetUp()
+    {
+    	FrameBufferApi::getInstance().impl = &frameBufferApiImplMock;
+    }
+
+    virtual void TearDown()
+    {
+    	FrameBufferApi::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(ScreenCaptureFixture, RegisteredMethods)
+{
+    Core::ProxyType<Plugin::ScreenCapture> plugin(Core::ProxyType<Plugin::ScreenCapture>::Create());
+    Core::JSONRPC::Handler* handler(&((Core::JSONRPC::Handler&)(*(plugin))));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler->Exists(_T("uploadScreenCapture")));
+}
+
+TEST_F(ScreenCaptureFixture, FrameBufferUpload)
+{
+    FBContext fbcontext;
+    //VncServerFramebufferAPI *api = nullptr;
+    PixelFormat pixelFormat = {32, 24, 0, 1, 255, 255, 255, 16, 8, 0};
+    vnc_uint8_t* frameBuffer = (vnc_uint8_t*) malloc(5120 * 720);
+    
+    Core::Event uploadComplete(false, true);
+    
+    EXPECT_CALL(frameBufferApiImplMock, fbCreate(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext** fbctx) {
+                *fbctx = &fbcontext;
+                return ErrNone;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbInit(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx, VncServerFramebufferAPI* server, void* serverctx) {
+                //api = server;
+                if (nullptr != server)
+                {
+                    server->framebufferUpdateReady(&fbcontext);
+                    server->framebufferDetailsChanged(&fbcontext, nullptr, 1280, 720, 5120, nullptr);
+                    server->paletteChanged(&fbcontext, nullptr);
+                    server->logMsg(&fbcontext, "msg");
+                }
+                return ErrNone;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetPixelFormat(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx) {
+                return &pixelFormat;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetWidth(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](FBContext* fbctx) {
+                return 1280;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetHeight(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](FBContext* fbctx) {
+                return 720;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetStride(::testing::_))
```

Comment:
please prefer ON_CALL like suggested in [readme](https://github.com/rdkcentral/rdkservices/tree/sprint/22Q4/Tests#faq). Instead of ::testing::Invoke you may use ::testing::Return